### PR TITLE
fixes #17062 - move master rack restart to puppet::server::service

### PR DESCRIPTION
**Pull request by domcleal**
Includes minor changes to containment of server configuration classes
to ensure notifies of changes from puppet::server::passenger and ::rack
all reach puppet::server::service.

### CODE REVIEW

File: manifests/server/install.pp [hunk 1]
```diff
@@ -31,6 +31,17 @@
     }
   }
 
+  # Prevent the master service running and preventing Apache from binding to the port
+  if $::puppet::server::passenger and $::osfamily == 'Debian' {
+    file { '/etc/default/puppetmaster':
```
Comment (ekohl) [inline manifests/server/install.pp]: It feels like this should be in ```config.pp```?

Comment (domcleal) [inline manifests/server/install.pp]: Yes, I'd agree except that I think it has to run before the package installation, or it needs reimplementing into debconf (if possible, I've not checked).

It used to be in puppet::config::passenger but had a `before => Class['puppet::server::install']` relationship, which caused a loop when I contained the puppet::config::passenger class. I had to move it here to better match the relationship it had.

Comment (ekohl) [inline manifests/server/install.pp]: I had a feeling it was something like that.

Comment (domcleal) [inline manifests/server/install.pp]: Sorry, I just noticed that I specified the relationship back to front, which may have confused matters. It now specifies that the config is _before_ the package.

**Comment by mmoll** [comment]
merged, thanks @domcleal!

**Pull request merged**